(Updated on 2021.12.29) Many changes Can't Title Summarize

SoraHjort opened the pull request:
tl;dr: many changes were made, please review changes and test stuff out before going with a merge. 

(Due to a woops I made at one point, it's being pulled from a new sub-branch. The actual full change history log: https://github.com/SoraHjort/webcomic_reader/commits/Fixes-and-Additions/webcomic_reader.user.js )

This is about a year in the making (on and off a lot). It shouldn't have taken that long, but I kept getting side tracked, fixing comics I was reading at the time, forgetting about it, and so on.

This started off as a branch to just add and fix comics before putting in pull requests. It then went into adding several features and other stuff. I am stuck in a "what do I do next" slump, which is why I'm putting it out now. I think I'm mostly done. But I would like feedback on the changes and the like.

If it comes down to it, and the changes I've made are not acceptable, I could separate the additions/fixes for sites.

Highlights: 
1. Styling to prevent sites from screwing things up (like with the settings window)
2. Fullscreen Button
3. (Mobile) Disable/Enable this Script button at the end of the page (blame greasemonkey on mobile for the idea)
4. `@includes` have been updated to be "http*://" instead of http and https to
 a. save space when it comes to duplicates
 b. future proof in case a domain goes from http to https
 c. more strict capture than simply going "*://" to prevent running on, say, ftp:// or any other protocol.
5. primarily a mobile issue, but script no longer run when viewing just a image files.

Please give this a good look over and test it out. I'm the kind of person to be paranoid over my own work when it's this big of a change. I worry I might have broken something, despite not running into any issues in the months I've been using my dev branch exclusively). Heck I'm still paranoid if I'm doing pulls correctly.

Thankfully, I kept notes on note worthy changes( IE: not every time I had to edit because I goofed)
The full change summary: 

(Note on Site fixes and added sites, if they've changed since I've made those changes, and no longer work, sorry for the inconvenience.)

 
Sites Fixed: 
- Homestuck (Hopefully)
- Bog and George (Multi Image and commentary having images issue)
- Girl Genius (Multi-image should be working just fine now!)
- Vicki Fox
- Free Fall  (Broken on the first 300-ish pages)
- Dominic Deegan (Partially for mobile users)
- Dilbert
- Lackadaisy Cats
- Kevin and Kell (could use some styling fixes on mobile when not minimalistic)
- Mangaseeonline (accidentally removed when it just needed a http* fix)
- DMFA
- LFG
- GrrlPower
- Keychain of Creation
- Three Panel Soul (Double Check later)
- TwoKinds
- Project Future (back would got ot the start of chapters instead of the previous page)
- FreeFall (minor, some pages had two comic images)
- Devil's Panties (Multi-image pages)
- VGCats (It's back, new site layout, new comic page)
- Misfile
- Furaffinity now works on both www and non www urls

Sites Added: 
- Yosh Saga
- Artificial Incident
- Kemono Cafe
- Comic Host Xepher
- Skin Horse
- Narbonic
- Sailor Sun
- Jeanie Bottle
- Dreamland Chronicles
- Snafu Comics (main issues are built into the archive, like pages being out of order, nothing we can do)
- Alice and the Nightmare
- Not a Villian
- Mac Hall
- e621.net
- Alien Dice
- Most if not all the other Shivae comics as well
- Xepher comics in general

Site tweaks:
- Furaffinity tries to default to comic navigation options. If it fails to find a next/back comic page, it'll default to gallery viewing mode
- DHS Comics @includes have been condensed into a singular one. 
- Various sites had their commentary set to also be nav enabled, reverted this change.

Ahab's White Whale: Apple Geeks, it doesn't deserve to be, but it's doing it's darnest to be it.

Features and Big Changes:
- Full screen button. Primarily useful for Mobile users, but who knows what desktop users could use it for!  https://youtu.be/TKl02m4lkJc?t=88
    > Made it so that it'll detect when fullscreen is exited, regardless of method of doing so.
- (Mobile only) A Disable/Enable This Script button at the bottom of the page. Greasemonkey on mobile already does this, as it inserts menu commands as a button. This makes it universal regardless if it's greasemonkey or tampermonkey. As well as not a tiny button for mobile users.
- Script's buttons have a default style
   > First and Last links are now buttons. Lightblue for first, Light red for last
- Various Colors (buttons, settings menu) fixed to not be affected by OS's color scheme
- Several more color and styling fixes
- Script now does not load on image files (png, JPG, gif, etc). This was mainly an issue on mobile from what I've seen.
- Non-Dynamically Colored Buttons now get darker when pressed.
- The Dimming background now overextends width and height to fix an issue on mobile when it came to scrolling
- Pages Dropbox now have a max size of 100% (not all browsers have been updated to allow min()/max() yet, otherwise it would be 500px or 100%)
- Settings window has had various updates to give a more uniform size and feel across each page. 
    > By also forcing elements within the settings window, it helps prevent a site from overriding it and making it a mess.
    > As well as work to help to prevent drifting that occurred on some sites.
- wcr_ext_navi_ctrls: a function that can be called in js: for automatically turning the extra section into a navigable part.
    > ex `js:	 wcr_ext_navi_ctrls`



- All includes have been future proofed by switching from HTTP and HTTPS to HTTP*. Should prevent the script from breaking if a site switches between them.

Sites fixed due to switching to "http*" (checked back in october/november 2019)
biggercheese.com/
bittersweetcandybowl.com/
bringbackroomies.com/
buckocomic.com/
https://www.c.urvy.org/
collectedcurios.com/
doctorcatmd.com/
dynasty-scans.com/
garfieldminusgarfield.net/
giantitp.com/
gpf-comics.com/
little-gamers.com/
meekcomic.com/
mrlovenstein.com/
pebbleversion.com/
pointguardian.com/
project-apollo.net/mos/
satwcomic.com/
scarygoround.com/
sevensmith.net/chirault/ - Front page broken
supercrash.net/
vgcats.com/
webtoons.com/ - Slightly broken, pages don't show up till you scroll down a little (fixed? I didn't do anything, but it hasn't happened in a while)



-Post note-
 Ahab's White Whale: Apple Geeks, it doesn't deserve to be, but it's doing it's darnest to be it.
I tried, many, many, many times to get it to work with it to no avail. It's... it's weird. Maybe one day I'll try again, but if someone wants to give it ago, go ahead. It's like it ignores the capture rules and instead wants to grab the banner image as "Back", despite the xpath being completely different.



There are also other ideas I do want to implement at some point, but I need to land on stable ground, script wise, before progressing on those fronts. Sorry it took nearly a year to get this going, but, well... those changes. And procrastination.

SoraHjort committed:
Update webcomic_reader.user.js

Fixed EGS Comics pre-loading.

SoraHjort committed:
Update webcomic_reader.user.js

Just a minor change to change the version/date

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Fixed:
Kevin and Kell
Dominic Deegan
GrrlPower
Keychain of Creation
Three Panel Soul

Major Changes:
Converted HTTP and HTTPS matching to HTTP* to future proof.
This may have fixed several sites previously noted for being broken.
This was the case for atleast Keychain of Creation.
May have to go over the broken list again.

SoraHjort committed:
Update webcomic_reader.user.js

(no real changes made, just wanted to list what sites did get fixed in the last change)
Sites fixed due to switching to "http*"
biggercheese.com/
bittersweetcandybowl.com/
bringbackroomies.com/
buckocomic.com/
https://www.c.urvy.org/
collectedcurios.com/
doctorcatmd.com/
dynasty-scans.com/
garfieldminusgarfield.net/
giantitp.com/
gpf-comics.com/
little-gamers.com/
meekcomic.com/
mrlovenstein.com/
pebbleversion.com/
pointguardian.com/
project-apollo.net/mos/
satwcomic.com/
scarygoround.com/
sevensmith.net/chirault/ - Front page broken
supercrash.net/
vgcats.com/
webtoons.com/ - Slightly broken, pages don't show up till you scroll down a little

SoraHjort committed:
Update webcomic_reader.user.js

Fixed LFG
Added KemonoCafe

Known Todo Currently:
CAD Comic
Curtailed Comic
Dilbert
Lackadaisy Cats
DMFA
Vicki Fox
Homestuck
Three Panel Soul - Front Page breaks the script.

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Let me tell you about Homestuck

Fixed DMFA/MissMab, Curtailed, and initial Homestuck work

When Homestuck changed to the new domain, it also got a revamped site. it no longer requires the multiple entries for detection depending on various things like which act we're in. This is good.

It however decided to be a pain to setup for whatever reason. Took several hours but I managed to get it finally.

It properly shows the image, the Pesterchat, and even the nav link. But since the link wasn't automatically switching to being #next, I used some JS to make it like the image and navigating by clicking left or right on it's box.

The other comics on the site, like Problem Sleuth aren't setup yet. That's on the way.


Also AppleGeeks is being a big pain in getting to work. Two days (on and off) have been wasted trying to get that to work.

Also think I forgot several commits ago to mention I fixed some Kevin and Kell aligning problems, particularly on smaller screens like phones.

Current List of To Fix:

Yosh Saga
Artificial Incident
Dilbert
Lackadaisy Cats
Vicki Fox
Three Panel Soul (minor fix due to the front page)
The various sites that have been moving off of Katbox that aren't ending up on Kemono Cafe
Two Kinds

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

More Homestuck. I had previously remembered to capture the pesterchat and nav stuff, but forgot the story text that sometimes up

SoraHjort committed:
Update webcomic_reader.user.js

Fixes to some alignment styling for Kevin and Kell

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Added Yosh Saga and Artificial Incident
Fixed Vicki Fox

Previous comitt: Added Fullscreen. Primarily for Mobile Users to clear up realestate. 
Examples: 
https://i.imgur.com/09x8xMJ.png
https://i.imgur.com/ThrFZfg.png

Also note for mobile users: It may be best to either enable "Request Desktop site" in the browser, or enable Fit-to-Screen with shrink width to fit enabled. Maybe on newer phones it isn't an issue, but on a Samsung S4 I've been testing this on, it likes to display pages zoomed in. And while in Fullscreen you can't zoom out.

SoraHjort committed:
Set theme jekyll-theme-hacker

SoraHjort committed:
Delete _config.yml

SoraHjort committed:
Update webcomic_reader.user.js

Solved: 
Hopefully fixed one or two issues with Dominic Deegan. Mainly a mobile issue, will have to check back after upping this.
Kevin and Kell had a minor issue with detecting first and last.

Back up and running:
Dilbert
Lackadaisy Cats

Added in default styling for buttons and dropdown. This should solve any issues where the site would override the buttons and what not to be obscenely large, small, or just didn't look like buttons. Coloring was also applied in the event that your system's color scheme made the text unreadable.

SoraHjort committed:
Update webcomic_reader.user.js

Nothing really. still trying to get applegeeks to work, it doesn't want to detect next and previous... This just contains the latest edit, incase i ended up working on it this weekend.

SoraHjort committed:
Update webcomic_reader.user.js

Setup default appearances for settings buttons. as well as setting a width to the settings dialog.
Need to remember later to set the default font and size later.

Also minor addition to Kevin and Kell, listing the page's title as an extra. Which appears to have been in the original setup. Contemplating if I should add in the comments section to the extra section as well. Might test it to see how much of a problem it causes since it's using Disqus.

Added in the ability to browse FA's galleries if they weren't on a comic page. Though if a comic page that uses the built in comic nav system hits the queue it will start going into that comic's pathing. Might have to look into some JS to get it to go "if I started in viewing Gallery images, stay in gallery navigation". It's a minor issue that might be a little annoying.

Also added the native gallery queue to the extra. Will work on styling for it later.

....still no luck on AppleGeeks. What I have SHOULD be working from what I can tell. It's just... not....

SoraHjort committed:
Update webcomic_reader.user.js

Well the Kevin and Kell extra change from last comitt was a bust. Because the title also contains a bit of javascript to detect how many comments are made on that page, it causes the image to have a lag time when switching pages.

Also found out that the "Disable on this site" button also only appears on mobile, maybe even just for the mobile version of greasemonkey. Will look into this more with tampermonkey as well. Would like to style these buttons, but they only have text and nothing else to ID them. And since CSS doesn't have a selector for what a tag contains between it... I even have to wonder if Javascript can touch the buttons since they should technically be added only after the script has ran.

I thought of several solutions, but they have their down sides
1. Default style for all buttons.
-Problem: This affects all buttons on a site
2. Use nth last child of type css selector to select the last button
-Problem: If this is ran outside of mobile, this will affect the last two buttons on a page.
3. Appended a hidden button towards the end of the page via script
-Problem: Too much effort for a minor thing.

These three are not ideal.

SoraHjort committed:
Update webcomic_reader.user.js

Testing to see if mobile and userscript engine detection stuff

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

A lot of these commits today will be related to trying to deal with the appearance of the buttons greasemonkey puts in on mobile for script menu command entries.

SoraHjort committed:
Update webcomic_reader.user.js

Primarily so I have a easy way to install and debug the script on mobile

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Fixed Free Fall. It wasn't detecting the comic page for the first 300-ish pages in the archive.

Also styling for the GreaseMonkey "disable for this site" has been implemented.
Will look into manually including the button for Tampermonkey users later.

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Added homestuck page titles to extras.
Testing another size alignment for Mobile greasemonkey buttons.

SoraHjort committed:
Update webcomic_reader.user.js

Would help if I had the css on the inside of {} instead of behind the }

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

And the +...

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Ok, this should solve that. For Greasemonkey on Mobile, it'll change the script command button to disable for the site to be larger and centered. 
Will look into manually adding a similar button for Tampermonkey and others for mobile.

SoraHjort committed:
Update webcomic_reader.user.js

Hopefully... This should finalize fixing up homestuck. Ran into an issue with pages with multiple images (as tends to be the case). Got it setup to take care of that... hopefully... I did kinda start rereading it, so i'll probably see if there are any problems that crop up.

SoraHjort committed:
Update webcomic_reader.user.js

Reverted a removal from dead site cleanup. mangaseeonline.us is back on.

SoraHjort committed:
Update webcomic_reader.user.js

Should fix the multi image problem on Bob and George. As well as pages where it was loading the commentary artwork as the comic image.
Also included the commentary within the extra capture (which is probably what fixed the commentary artwork issue)

SoraHjort committed:
Update webcomic_reader.user.js

It's messy looking if the script first loads on it, but this should fix the one(?) page in Girl Genius that has multiple images. Gave it additional leeway in the event they do that again, but with more images (up to 10)

Same can be said about Bob and George from the previous revision.

SoraHjort committed:
Update webcomic_reader.user.js

... Forgot to make it so that the multipage fixes for B&G and GG were "navigatable" by clicks

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

More work on the enable/disable buttons for mobile users

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

That should take care of that.

Mobile Users
Greasemonkey - Resizes the Enable/Disable button at the bottom of the page (the button is automatically added as GM adds script commands as buttons)
Tampermonkey - Adds in the button in roughly the same location.

This makes it easy for mobile users to enable and disable the script on a site without having to go through menus. (Or zooming in on a tiny button in GM's case)

SoraHjort committed:
Update webcomic_reader.user.js

Just found out one of the changes I did a while back is making the "disable script" button that shows up for mobile users using tampermonkey to.... reproduce......... hopefully this'll fix it. Can't really test it without uploading it, because while I can get a mobile view setup on desktop to test these things, you can't externally edit scripts for tampermonkey.

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Ok, previous fix did fix the problem. Had the function being called for every page cache. I fixed it in a not-so elegant manner though.. It's embedded in the check on whether the script will put in "enable this script" or "disable this script" checks for script commands.

Also this contains a minor fix for Homestuck. It'll make it so any pages with a movie will not be cached. It was trying to cache those pages which resulted in no video, as well as the need to disable the script to view said movie files.

SoraHjort committed:
Update webcomic_reader.user.js

Fixes for Homestuck, and added Homestuck^2
Still need to add in the old comics for MSPaint Adventures

SoraHjort committed:
Update webcomic_reader.user.js

Version number update to get tampermonkey to not be mad over last change.

SoraHjort committed:
Update webcomic_reader.user.js

Added the ability to favorite submissions while the script is enabled for FA

SoraHjort committed:
Update webcomic_reader.user.js

Minor fix for Homestuck. And will need to revisit soon to grab content_container for it, primarily because of homestuck2

SoraHjort committed:
Update webcomic_reader.user.js

Fix for Homestuck 2, as well as possibly fixing any issues on homestuck i hadn't run into yet in my slow-reread.

SoraHjort committed:
Update webcomic_reader.user.js

Found out that MSPaint Adventures is still up for those who want the old flash [s] archives. Regression and url edit to add the support back in. Or atleast most, not sure about the "extra" url, will try to track that one down.

SoraHjort committed:
Update webcomic_reader.user.js

More adjustments to Homestuck
- Made it clear that there was a flash version on pages if that warning popped up (won't load it but will display the message more clearly if set to dimmed)
- Changed the Nav link for next to be strictly for next. As well as making it so that the cursor will display the next cursor regardless which side you're hovering.

SoraHjort committed:
Update webcomic_reader.user.js

Another fix for Homestuck. Testing an option, though don't know if it works. 

Ran into an issue with capturing when during my reread I hit https://www.homestuck.com/story/3764
Will look into it for a fix

SoraHjort committed:
fixed issue with squished images

It seemed randomly, but some sites would render with a squished or stretched image. Turns out, it's likely due to the site setting "100%" to max-width or max-height. This should fix that for most if not all sites. Just needed two lines to set the max sizes in the same section where it sets the width and height to be [native or scaled]

SoraHjort committed:
Update webcomic_reader.user.js

Update for homestuck. Worked out a temp workaround. Will have to come back to clean it up. 
https://www.homestuck.com/story/3857
https://www.homestuck.com/story/63

SoraHjort committed:
Update webcomic_reader.user.js

Fix for multi-image pages for Devil's Panties (it's not satanic porn, seriously!)

SoraHjort committed:
Update webcomic_reader.user.js

Updated EGS Code based on recent pull merge on main

Rule Excluded: PNG JPG JPEG and GIF
On mobile it would force the script's UI to load if you were viewing the image standalone.

Tried to fix missing image from http://www.girlgeniusonline.com/comic.php?date=20111104
No luck, it's for whatever reason detecting the first and third images only.

Homestuck
-Hopefully fixed Scratch storyline's background issues

-It's still a little bit of a mess, both code and what renders. In some sections it'll double the story text and navigation (in two different styles) due to the different detection methods.

SoraHjort committed:
Update webcomic_reader.user.js

Testing the Update URL function

SoraHjort committed:
Update webcomic_reader.user.js

Replaced some spaces with tabs

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

Additional Fixes for Homestuck, primarily when it comes to minimalistic mode.

SoraHjort committed:
Update webcomic_reader.user.js

Added/fixed Snafu comics. The script mostly worked with it, without needing to set up a custom entry for it. However back was forward and forward was back in terms of navigation. So that should hopefully fix that as well.

SoraHjort committed:
Update webcomic_reader.user.js

Minor fix for Snafu Comics. Had an annoying resize transition effect going on.

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Update webcomic_reader.user.js

First pass at adding Narbonic

SoraHjort committed:
Update webcomic_reader.user.js

Fix for Narbonic, needed some detection tweeks.

SoraHjort committed:
Update webcomic_reader.user.js

Added Dreamland Chronicles. (Was once a old and completed comic, but they've recently started remaking old chapters that were originally done in CGI with the hand drawn art in the later chapters. Sadly, they've taken down that old archive)

SoraHjort committed:
Update webcomic_reader.user.js

Minor Fixes for:
Narbonic - Force a whitebackground for the random images with transparent background, causing it to be hard to read when dimmed.
GrrlPower - Removed comment section from extras, longer load times between pages
Xepher.net - Webcomic Host - Whitelisted. Works out of the box for several comics, others will need to be manually fixed.

SoraHjort committed:
Update webcomic_reader.user.js

Added:
Skin-Horse.com
JeanieBottle.com
SailorSun.org

SoraHjort committed:
Updated to latest pulls

Updated to include the following
Mangasee - https://github.com/anka-213/webcomic_reader/pull/150
CSS/XPath Comment fixes - https://github.com/anka-213/webcomic_reader/pull/149
Awkward Zombie - https://github.com/anka-213/webcomic_reader/pull/148
Arrow Symbols - https://github.com/anka-213/webcomic_reader/pull/147

SoraHjort committed:
Freefall Fix

Fixes for the rare occurrence of multiple images in the archives.

SoraHjort committed:
Twokinds

Fixes TwoKinds
grabs the transcript and the sketch as extras

SoraHjort committed:
Project Future

Fix for Project Future: back was going to the start of previous chapters.
(Project Future was automatically added when the host xepher got added as whole)

SoraHjort committed:
Version 2020.07.21-150200

just adding a version number

SoraHjort committed:
More UI updates/fixes/changes

Back at work at trying to update the settings menu to be more universal and eye appealing. As well as preventing sites from screwing with it with their own CSS.

SoraHjort committed:
Fixed TwoKinds, for real this time... probably

There was a slight issue with TwoKinds early archives. this fixes this hopefully (will be doing a re-read soon)

...Also made me reason that a lot of stuff I and others in the past have been doing is slightly wrong.. a lot of // where a / would work just fine and prevent issues. May have to give a look at the various entries and check and cleanup.

SoraHjort committed:
Twokinds Fixed Take 3

Something something twokinds, something something tampermonkey specific issue, something something doesn't like specifying data type in object tags.

SoraHjort committed:
More UI stuff

SoraHjort committed:
UI and small fixes and cleanup

Buttons: Now change color when pressed. Doesn't work with dynamically colored buttons currently
Cleanup: Set repeatable CSS code for Disable Script Button to be a variable to save space
UI: more size fixes. Removed New lines on things that can fit on one line to give a more uniform appearance among list entries.

LFG Banner is fixed
Applegeeks got removed, I've given up for the time being.

(might be more, forgot what else I might have done)

SoraHjort committed:
wcr_ext_navi_ctrls + Twokinds Extra Navigation Fix

Added a function that I've reused multiple times to save space.
By just having "js:	wcr_ext_navi_ctrls" in an entry, it will make it so that anything in the "Extra" section will be counted as navigable like the primary page. This will help somewhat on pages that have multiple images. Just have the extra images be caught by "extra:" and throw that in at the bottom of the comic's entry, and bam!

The reason for this is that the same function has been used multiple times in the past. It wasn't a dedicated function, thus it was code that was being repeated several times. Thus: single function that could be called anytime, presto!

This won't work for every instance, for an example Homestuck. Due to the way that one is setup in the capture, it's got it's own edited version of this function running.

Also twokinds' extra now is now navigable as well.

SoraHjort committed:
Girl Genius Fix

Fixes a long standing issue on page http://www.girlgeniusonline.com/comic.php?date=20111104
hopefully it doesn't break any other multi-image pages. Tests show that it should be fine, but without a full reread of the archive, I can't be 100% certain.

SoraHjort committed:
2020.07.27.160900 - Exit Fullscreen Fix

An outstanding issue with the fullscreen function was if you left fullscreen from any other method than the added button it would require you to press the button several times or reload the page. It should now detect when fullscreen has exited and change the varialbes as well as the text on the button to reflect it. So it should now be able to go back into fullscreen easily.

SoraHjort committed:
Updated to 2020.07.27.160900 from Fixes and Adds

See https://github.com/SoraHjort/webcomic_reader/issues/1#issuecomment-541385418 for a summary of changes

Didn't do a merge/pull from Fixes and Additions, because I did something silly by accident with it quite a few months back. Accidentally added an "environment" and it's been doing weird things since with it. Thus, I just copied and pasted it here. 

The Full Changelog Edit History can be found here: https://github.com/SoraHjort/webcomic_reader/commits/Fixes-and-Additions/webcomic_reader.user.js

SoraHjort committed:
Trying to get master branch to match up with main

step 1

SoraHjort committed:
Merge pull request #2 from anka-213/master

test pull

SoraHjort committed:
Fixed some spacing

During my "Spaces to Tabs" cleanup, it changed some double spaces into tabs that it shouldn't have. Particularly ones after : 
This'll hopefully clean some of that back up.

SoraHjort committed:
Merge pull request #3 from SoraHjort/master

Hopefully fixing compares and commits with PR Branch

SoraHjort commented:
One of these days I'll properly learn and deploy pulls the first time. Maybe. 
That should fix any of the odd commits/compares.

And it looks like the silliness/woops I mentioned carried over onto the main branch, for some reason. Basically I accidentally created an "environment" and couldn't quite figure out how to remove it. Hence the parts where it says "set theme" and "delete _config.yml"

I'll contemplate if I should just do a "merge" from my fixes and additions branch with pull-requester so it has the full commit history. Since what I was trying to avoid happened anyway.

SoraHjort committed:
2020.07.28.110700

anka-213 commented:
You can look at how to do "interactive rebase" `rebase -i` (followed by a `push -f`) if you want to fix it. But it is also fine to leave it as it is.

SoraHjort committed:
Merge branch 'Fixes-and-Additions' into master

SoraHjort committed:
Merge pull request #4 from SoraHjort/master

getting commit history transfered over Pt.1

SoraHjort committed:
Merge pull request #5 from SoraHjort/Fixes-and-Additions

getting commit history transfered over Pt.2

SoraHjort commented:
Too late, fixed it :P
I do apologize though, I am so not used to these systems. I can manage with personal edits and what not. But pulls are still foreign to me. (Also that environment mixup threw me through a loop trying to undo)

anka-213 commented:
No problems :)

SoraHjort committed:
Undo Accidental Reversion on XPath Capitalization

Notice while re-reviewing the compare list that one section of the XPath correction put forth by roryokane got undone.
Also undid my last one as I noticed it put a lot of things out of line, more so that it should have.
*headdesk*

SoraHjort committed:
2020.07.28.115100 - Tab Alignments

(I should really stop making commits after I've submitted a pull request, but I kept finding stuff, hopefully this is the last)
Minor: Went through and cleaned up some alignment stuff via tabs indents. Found some spaces I missed prior, gave a glance at things to make sure they aligned in a uniform fashion.

SoraHjort committed:
2020.07.30.130000 - Minor Fixes/clean up

Realized that I could do the exclusion on image files a bit better. So now it'll run on the rare site that decides to have something like index.php?file=thisimagefile.jpg (see Fur Will Fly for an example of this listing style (FWF can't be captured eitherway because the site forces HTTPS protocol in the URL, but doesn't actually transmit via the standard))
`// @exclude		/^(?!.*(?:\.html|\.php|\?|\=)).*(jpg|jpeg|png|gif)$/`

Also shrunk the includes and captures for dhscomix with some regex. Not only to save space, but also to auto-capture new subcomics they create.
`// @include		/^https?://(www\.|)dhscomix.com/.*comics/.*$/`

SoraHjort committed:
2020.08.03.152600 - Girl Genius&EGS extra captures

Redid Girl Genius's Extra capture. Didn't like how it didn't scroll. And after several attempts at creating a variation of the extra navigation function to do auto scrolling (scrollear() prevents this), I redid the capture setup. It resulted in less space used.

Also, during a re-read of EGS, I noticed that every so often Dan would have a link that went to a "alt" image. I put in a capture for that to display it (at 50% width to differentiate it in cases where he's showing off that he made an edit, and thus not-canon)

SoraHjort committed:
2020.08.14.134000 - EGS, Girl Genius, & Machall

EGS and GG's extra captures got updated.
EGS's needed it's last update duplicated to it's www variant
GG's now is set to be an inline-block to fix issues in minimal. As well as making the news/commentary notes actually visible due to some CSSing that occurred

Also I added MacHall to the captures. It's a long finished comic from ages ago. It was part of the big old school trio (MacHall, AppleGeeks, & MegaTokyo). Still no word on getting applegeeks captured....

(I realize I keep adding stuff to the pull that I put up. Atleast I'm limiting it to once every several weeks)

SoraHjort committed:
Forgot to update EGS's variant's style. woops.

SoraHjort committed:
Some changes, testing for mobile.

SoraHjort committed:
Update webcomic_reader.user.js

SoraHjort committed:
Alien Dice

Alien Dice added

SoraHjort committed:
Forgot to update version number

SoraHjort committed:
Minor Change, trying something new

Trying something new, since I'm gonna go through Alien Dice's archive, I decided to throw in some extra catches for the comic image detection. In the off chance there are pages with multiple images. I don't recall there being any such pages in the archives, but it is something that I wanted to test out anyway. And become a common practice with me on sites that I add personally.

SoraHjort committed:
GG minor fix and Shivae comics

Minor fix (non-ideal) for the "latest page" of Girl Genius.

Added the rest of Shivae Comics, as well as a minor fix for Alien Dice.

Still have to look into several issues with these sites, primarily the lack of overflow scrolling, as well as the sites managing to color the setting window's buttons. Would take care of it now but I barely got the time for it right this minute.

SoraHjort committed:
2021.12.22.154800

Comics added:
Added Alice and the Nightmare

Comic Fixes:
Fixed Misfile
furaffinity non-www pages
smackjeeves's @include removed (defunct). Left capture rules for use within the internet archive
e*hentai's body 2px padding removal

Other Fixes:
max width on recent/loaded pages drop box

SoraHjort committed:
2021.12.28.013500

.... Properly fixed the pages dropbox this time, forgot to move a bracket

Also converted First and Last into buttons to make it more visually cohesive.

SoraHjort committed:
2021.12.28.014000

Testing fix for dimming on mobile not properly covering when scrolling.

SoraHjort committed:
2021.12.28.015900

Realized that NAV Comic never got added. 

Side note: seems the dimming fix has mostly worked. Particularly on minimalist mode, the only issue I'm seeing is on some sites where they have a background image (artificial incident for an example) the background image still has the same issue. But that may be less of an issue due to WCR outside of minimalist view.

SoraHjort commented:
So finally working on this again (in a branch which I'll merge into this pull request)

The problem I'm having right now is that the english translation... made a bit of a mess of things.

Not on their part, mind you. It's gone and created a lot of conflicts that I have to go in manually.

I thought I could just use github's desktop app to rebase and what not... buuuuuuuut! It's not great. It doesn't even use the complete history of changes I made in the branch I'm working in, only up to July of last year, for some reason.

So, what I'm basically going to have to do is manually replace all the translation stuff into my fork. 

Plus it'd probably be a lot quicker than having to go through the 100+ changes it wants me to check every, single, time. It wanted to open every single conflict in my editor. A lot of them were going from four spaces to tab spaces. Though some of them are related to functions and new sites.

When I'll get this done, I do not know. I'll basically have a week free from everything, because I just moved, the ISP tech is swamped and can't setup the lines till next week at the earliest. And with the power of not-hotspotting with a smartphone, the only thing I can really do with my internet connection is browse the web, watch videos, and play games I've already downloaded.

So yeah, got free time for that now, minus some more moving related stuff.

But yeah, got a bunch of stuff to add to this pull, fixes, visual cohesion stuff, new sites, and what not in the works. Trying to keep myself from adding more stuff till I get the translation stuff pulled in.

Again, sorry it's taken so long to get around to doing this.

SoraHjort committed:
2021.12.28.225600

Had to semi-manually update my file to get it rebased to anka main. Came across GitKraken and it saved me hours, maybe even days, manually going through copying and pasting. Sadly due to a issue on my end (temporary internet connection related) I couldn't submit it via that. 

But this does bring it up to the main's english translation changes.

SoraHjort committed:
Merge branch 'Pull-Requester' into Fixes-and-Additions

SoraHjort committed:
Merge pull request #7 from SoraHjort/Fixes-and-Additions

Update to latest Fixes and additions Branch

SoraHjort committed:
Merge branch 'master' into Pull-Requester

SoraHjort commented:
Managed to get the translation inserted sooner (thanks to GitKraken. Found it earlier, not long after my last comment, it's been nice so far in it's small test). Still had to do some manual stuff. Saved me probably a few days of eye sore work.

Updated first comment to include the changes made. Not too much changed, some sites added, some sites tweaked, unified navigation buttons, some mobile related fixes (such as dimmed mode having issues while scrolling). 

Also resolved the github file conflicts that would come up, of course.

Things should be good to go for a merge in if deemed so. 

For the most part I've been using my own dev branch for the past year or so, so I can basically say that the major changes have been working just fine. And I haven't made much major additions/changes during that time due being dragged from one corner of the USA, to basically it's opposite, for a year.

It should be a while before I submit anything new to be pulled into the project. Such as more sites, as well as some feature tweaks. So there are things I still want to do, but that can wait for a later pull.

edit: also fixed this in the process https://github.com/anka-213/webcomic_reader/issues/158

anka-213 commented:
I’ll go ahead and merge this now so more people can get access to all your fixes
